brush-core/src/shell.rs

```diff
@@ -1506,8 +1506,11 @@ impl Shell {
     /// # Arguments
     ///
     /// * `command` - The command to trace.
-    pub(crate) fn trace_command<S: AsRef<str>>(&self, command: S) -> Result<(), std::io::Error> {
-        let ps4 = self.get_env_str("PS4").unwrap_or_else(|| "".into());
+    pub(crate) async fn trace_command<S: AsRef<str>>(
+        &mut self,
+        command: S,
+    ) -> Result<(), error::Error> {
+        let ps4 = self.as_mut().expand_prompt_var("PS4", "+ ").await?;
```

Comment:
From some quick testing, it looks like `bash` defaults the trace prefix to `""` if `PS4` is *unset*, e.g.:

```console
$ echo $PS4
+
$ set -x; echo hi; set +x
+ echo hi
hi
+ set +x
$ unset PS4
$ set -x; echo hi; set +x
echo hi
hi
set +x
```

(Do you see the same?) That tells me that we should pass `""` as the 2nd arg here. If `PS4` was unset on entry into the shell, we should still benefit from the `Shell` initialization code path that gives it a default value of `"+ "` if unset.

Reply:
Good catch!  fixed.